tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -11,20 +13,60 @@ def value_is_empty(value, length):
     return value is None or value in empty_values
 
 
+def generate_parser_error(datafile, line_number, schema, error_category, error_message, record=None, field=None):
```

Comment:
I think a generator function like this might solve our `set_status` problem in 1613